test/tools.py

```diff
@@ -269,6 +291,39 @@ def generate_test_repo(tmpdir, values=[0], dvcs_type='git',
     return dvcs
 
 
+def make_test_repo(tmpdir, dvcs_type, operations):
```

Comment:
Maybe I could adapt my `make_test_repo()` and refactor the test code to replace `generate_test_repo()`. Let me known.